qcodes/instrument_drivers/rohde_schwarz/ZNB.py

```diff
@@ -127,43 +81,27 @@ def set_sweep(self, start, stop, npts):
         self.setpoints = (f,)
         self.shape = (npts,)
 
-    def get(self):
-        if not self._instrument._parent.rf_power():
-            log.warning("RF output is off when getting mag")
-        # it is possible that the instrument and qcodes disagree about
-        # which parameter is measured on this channel
-        instrument_parameter = self._instrument.vna_parameter()
-        if instrument_parameter != self._instrument._vna_parameter:
-            raise RuntimeError("Invalid parameter. Tried to measure "
-                               "{} got {}".format(self._instrument._vna_parameter,
-                                                  instrument_parameter))
-        self._instrument.write('SENS{}:AVER:STAT ON'.format(self._channel))
-        self._instrument.write('SENS{}:AVER:CLE'.format(self._channel))
-        self._instrument._parent.cont_meas_off()
-
-        # instrument averages over its last 'avg' number of sweeps
-        # need to ensure averaged result is returned
-        for avgcount in range(self._instrument.avg()):
-            self._instrument.write('INIT{}:IMM; *WAI'.format(self._channel))
-        data_str = self._instrument.ask(
-            'CALC{}:DATA? FDAT'.format(self._channel))
-        data = np.array(data_str.rstrip().split(',')).astype('float64')
-        if self._instrument.format() in ['Polar', 'Complex',
-                                         'Smith', 'Inverse Smith']:
-            log.warning("QCoDeS Dataset does not currently support Complex "
-                        "values. Will discard the imaginary part.")
-            data = data[0::2] + 1j * data[1::2]
-        self._instrument._parent.cont_meas_on()
-        return data
-
+    def get_raw(self):
+       data = self._instrument._get_sweep_data()
+       if self._instrument.format() in ['Polar', 'Complex',
+                            'Smith', 'Inverse Smith']:
+           log.warning("QCoDeS Dataset does not currently support Complex "
```

Comment:
Perhaps this warning should point to the other parameter

Reply:
full name is 'Instrument name'_'Channel name' and short_name is 'Channel name'.
In my case full name is VNA_S21 for and short_name is S21.

Reply:
good point